Billing storage

Tbaut opened the pull request:
closes #1969
I left out the notifications so that it doesn't interfere too much with the design refactor #1968

---
Submission checklist: 

<!-- Remove anything below that is not applicable -->   

#### Layout
- [x] Change looks good in the desktop web ui
- [x] Change looks good in the mobile web ui

#### Theme
- [x] Components / elements inspected in light mode (there's no dark mode on Storage)

Tbaut committed:
without notif for now

Tbaut committed:
Merge branch 'dev' of github.com:ChainSafe/ui-monorepo into feat/tbaut-billing-storate-1969

Tbaut committed:
with storage plans

Tbaut committed:
fix routing

Tbaut commented on packages/storage-ui/src/Components/Pages/SettingsPage.tsx:
```diff
@@ -174,6 +178,16 @@ const SettingsPage: React.FC = () => {
             >
               <ApiKeys />
             </TabPane>
+            <TabPane
```
This needs to be rendered conditionally with `{isBillingEnabled ? <TabXYX /> : null}`

actions-user committed:
lingui extract

asnaith commented:
Just noticed a couple of issues

1) The link colour is the wrong shade of blue 
<img width="497" alt="Screen Shot 2022-02-25 at 2 39 56 PM" src="https://user-images.githubusercontent.com/3982665/155814745-c3f0a509-be52-4d49-83d9-157ef57a2f44.png">

<img width="518" alt="Screen Shot 2022-02-25 at 2 50 32 PM" src="https://user-images.githubusercontent.com/3982665/155814764-788fe405-06c6-4c37-9cd8-5946d35b10cb.png">

2) Add / update card modals are not working because the inputs are not getting the response from stripe to activate. 

I was able to test the upgrade and download flows successfully by adding a card to the profile via **Files** and then logging back into Storage. I had no issues upgrading with credit card or downgrading. I didn't pay via crypto but everything else with the flow and ability to return etc is all good.

Tbaut committed:
Merge branch 'dev' into feat/tbaut-billing-storate-1969

Tbaut committed:
proper blue

Tbaut committed:
update card modal full width

Tbaut commented:
Thanks Andrew for the review. I fixed the color issue for the links, and checked with the CC, from the Storage interface I was able to delete one (that I added from Files), and add one. Also note that you should make sure to have the Stripe pk env variable in you local environment. I just added to Render as it wasn't present. This should solve your issues.

Other than that things worked well for my tests.

tanmoyAtb reviewed (approved):
Great ! All the flows are working nicely ! 👍

asnaith commented:
>  Also note that you should make sure to have the Stripe pk env variable in you local environment. I just added to Render as it wasn't present. This should solve your issues.

Ah yes, completely forgot about the .env file! Thanks, I'll try again with that in place

asnaith commented:
I was able to test properly locally after adding the stripe key to the Storage UI .env 😅. 

The feature is working well, just two minor points remaining:

- The whitelist for the subscription tab access, at the moment all can access it
- The stripe key for render might need looking at (it still wasn't working for me)

Tbaut committed:
Merge branch 'dev' into feat/tbaut-billing-storate-1969

Tbaut commented:
I relaunched the deployment for render.. and checked, the env variable is definitely set :shrug: 

Pushing the whitelist check so that we don't forget it. Any reviewer can comment it.

Tbaut committed:
bring back restriction

asnaith reviewed (approved):
Yep, I can see the latest change has implemented the whitelist 👍

Render is still not working with stripe elements for me but approving as that has nothing to do with code in this PR.

FSM1 committed:
Merge branch 'dev' into feat/tbaut-billing-storate-1969

Tbaut commented:
I'll check the render issue more in details
